packages/databricks-vscode/src/configuration/ConfigModel.ts

```diff
@@ -38,22 +38,43 @@ export class ConfigModel implements Disposable {
             if (this.target === undefined) {
                 return {};
             }
-            const overrides = this.overrideReaderWriter.readAll(this.target);
+            const overrides = await this.overrideReaderWriter.readAll(
+                this.target
+            );
             const bundleConfigs = await this.bundleConfigReaderWriter.readAll(
                 this.target
             );
-            const newValue = {...bundleConfigs, ...overrides};
-
-            if (JSON.stringify(oldValue) !== JSON.stringify(newValue)) {
-                this.onDidChangeEmitter.fire();
+            const newValue: DatabricksConfigs = {
+                ...bundleConfigs,
+                ...overrides,
+            };
+
+            for (const key of <(keyof DatabricksConfigs)[]>(
+                Object.keys(newValue)
+            )) {
+                if (
+                    oldValue === null ||
+                    JSON.stringify(oldValue[key]) !==
+                        JSON.stringify(newValue[key])
+                ) {
+                    this.changeEmitters.get(key)?.emitter.fire();
+                    this.onDidChangeAnyEmitter.fire();
```

Comment:
Might be useful to move this out of the for loop, to not blast the same event many times (so emit it only once if at least something has changed)

Reply:
This is fixed in one of the later PRs https://github.com/databricks/databricks-vscode/pull/941/files